docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install

```diff
@@ -36,3 +53,84 @@ function hs_paragraph_types_update_9401() {
     $paragraph->save();
   }
 }
+
+/**
+ * Update postcard body field text format.
+ */
+function hs_paragraph_types_update_9402() {
+  // Load filter_format and editor exported configuration.
+  $config_directory = new FileStorage(Settings::get('config_sync_directory'));
+  $filter_format_config = $config_directory->read('filter.format.minimal_html_with_styles');
+  $editor_config = $config_directory->read('editor.editor.minimal_html_with_styles');
+  if (!FilterFormat::load('minimal_html_with_styles')) {
+    $filter_format = FilterFormat::create($filter_format_config);
+    $filter_format->save();
+  }
+  if (!Editor::load('minimal_html_with_styles')) {
+    $editor = Editor::create($editor_config);
+    $editor->save();
+  }
+
+  // Update field_hs_postcard_body field defnition exported configuration.
+  $field_config = \Drupal::configFactory()->getEditable('field.field.paragraph.hs_postcard.field_hs_postcard_body');
+  $field_config->set('third_party_settings.allowed_formats.allowed_formats.0', 'minimal_html_with_styles');
+  $field_config->save();
+}
+
+/**
+ * Update user permissions for minimal_html_with_styles text format.
+ */
+function hs_paragraph_types_update_9403() {
+  // Update contributor role permissions.
+  $role_config = \Drupal::configFactory()->getEditable('user.role.contributor');
+  $existing_permissions = $role_config->get('permissions');
+  $existing_permissions[] = 'use text format minimal_html_with_styles';
+  $role_config->set('permissions', $existing_permissions);
+  $role_config->save();
+
+  $role_config = \Drupal::configFactory()->getEditable('user.role.site_manager');
+  $existing_permissions = $role_config->get('permissions');
+  $existing_permissions[] = 'use text format minimal_html_with_styles';
+  $role_config->set('permissions', $existing_permissions);
+  $role_config->save();
```

Comment:
`user_role_grant_permissions` does all the work. use functions that exist.

Reply:
Fixed. This and the update dependencies let us run these update hooks during the D10 upgrade